File: Framework/CSharp/Framework/Framework/SmartGuid.cs
/*
 * 项目：Smartkernel.Framework
 * 作者：曹艳白
 * 邮箱：caoyanbai@139.com
 */
using System;

namespace Smartkernel.Framework
{
	/// <summary>
	/// Guid
	/// </summary>
	public static class SmartGuid
	{
		/// <summary>
		/// 判断是否是Guid
		/// </summary>
		/// <param name="guid">guid的字符串表示形式</param>
		/// <returns>判断的结果</returns>
		public static bool IsGuid(string guid)
		{
			Guid result;
			return Guid.TryParse(guid, out result);
		}

		/// <summary>
		/// 产生Guid
		/// </summary>
		/// <returns>结果</returns>
		public static string NewGuid()
		{
			return Guid.NewGuid().ToString("N").ToLower();
		}
	}
}

File: Framework/Java/framework/framework/smartkernel/framework/SmartGuid.java
/*
 * 项目：Smartkernel.Framework
 * 作者：曹艳白
 * 邮箱：caoyanbai@139.com
 */
package smartkernel.framework;

import java.util.*;

/**
 * Guid
 * 
 */
public class SmartGuid {
	/**
	 * 产生Guid
	 * 
	 * @return，结果
	 */
	public static String newGuid() {
		return UUID.randomUUID().toString().replace("-", "");
	}

	/**
	 * 判断是否是Guid
	 * 
	 * @param guid
	 *            ，guid的字符串表示形式
	 * @return，判断的结果
	 */
	public static boolean isGuid(String guid) {
		return guid.matches("[a-fA-F0-9]{32}");
	}
}
